app/main.py

```diff
@@ -1,6 +1,24 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cache_dict = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
```

Comment:
The return type hint for the `wrapper` function should match the return type of `func`. Since `func` is a `Callable`, the return type of `wrapper` should be the same as the return type of `func`, not `Callable`. Consider using `Any` or the specific return type of `func` if known.